lkql/extensions/src/lkql-evaluation.adb

```diff
@@ -436,19 +418,20 @@ package body LKQL.Evaluation is
       Result  : Boolean;
       Left    : constant L.LKQL_Node := Node.F_Left.As_LKQL_Node;
       Right   : constant L.LKQL_Node := Node.F_Right.As_LKQL_Node;
-   begin
-      case Node.F_Op.Kind is
-      when LCO.LKQL_Op_And =>
-         Result :=
-           Bool_Eval (Ctx, Left) and then Bool_Eval (Ctx, Right);
-      when LCO.LKQL_Op_Or =>
-         Result :=
-           Bool_Eval (Ctx, Left) or else Bool_Eval (Ctx, Right);
-      when others =>
-         raise Assertion_Error
-           with "Not a short-circuit operator kind: " & Node.F_Op.Kind_Name;
-      end case;
 
+      Left_Result  : constant Boolean := Booleanize (Eval (Ctx, Left));
+      Right_Result : constant Boolean := Booleanize (Eval (Ctx, Right));
```

Comment:
Unless I'm missing something, I think the "short-circuit" behavior was lost in this refactoring!